### PR TITLE
Sax handler

### PR DESCRIPTION
**Pull request by KrasimirRaikov**
Parsing an xml document into a list of employes, who have employer and address, each.

### CODE REVIEW

File: src/com.clouway.xml/Address.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
+package com.clouway.xml;
+
+/**
+ * @author Krasimir Raikov(raikov.krasimir@gmail.com)
+ */
+public class Address {
+
+    public static final class Builder {
```
Comment (panayotkulchev) [inline src/com.clouway.xml/Address.java]: The same suggestion as in the PR with for parsing xml with DOM, about if have 3 or less props use constructor.

File: src/com.clouway.xml/SaxHandler.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    @Override
+    public void startElement(String uri, String localName, String qName, Attributes attributes) throws SAXException {
+        try {
+            if (qName.equalsIgnoreCase(clazz.getSimpleName())) {
+
+                parent = clazz.newInstance();
+
+            } else {
+                Field[] fields = child == null ? clazz.getDeclaredFields() : child.getClass().getDeclaredFields();
+                for (Field field : fields) {
```
Comment (panayotkulchev) [inline src/com.clouway.xml/SaxHandler.java]: Can you get the field without iterate all? I commented in the other PR so me is engouhg to fix it in only one of them and see you understand my point.

Comment (panayotkulchev) [inline src/com.clouway.xml/SaxHandler.java]: If you decide to leave the for loop, you can `break` if you have match

File: src/com.clouway.xml/SaxHandler.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        } catch (IllegalAccessException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private Field getDeclaredFieldIgnoreCase(Object object, String fieldName) {
+        Field field = null;
+        for (Field field1 : parent.getClass().getDeclaredFields()) {
+            if (field1.getName().equalsIgnoreCase(fieldName)) {
+                field = field1;
```
Comment (panayotkulchev) [inline src/com.clouway.xml/SaxHandler.java]: here `break` after match too

File: src/com.clouway.xml/SaxHandler.java [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+                        child = null;
+                    }
+                }
+            }
+        } catch (IllegalAccessException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private Field getDeclaredFieldIgnoreCase(Object object, String fieldName) {
```
Comment (panayotkulchev) [inline src/com.clouway.xml/SaxHandler.java]: `private` methods stay under `public`

File: test/com.clouway.xml/SaxHandlerTest.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+ * @author Krasimir Raikov(raikov.krasimir@gmail.com)
+ */
+public class SaxHandlerTest {
+
+    @Test
+    public void happyPath() throws ParserConfigurationException, SAXException, IOException {
+        List<Employee> expectedElements = new ArrayList<>();
+        Address addressOne = aNewAddress().street("washington str.").streetNo(2).build();
+        Employer employer = aNewEmployer().name("US nation").startDate(Date.valueOf("1945-11-12")).endDate(Date.valueOf("1954-11-10")).build();
+        Employee employeeOne = aNewEmployee().firstName("George").age(56).address(addressOne).employer(employer).build();
```
Comment (panayotkulchev) [inline test/com.clouway.xml/SaxHandlerTest.java]: Another way to build composite objects that can be constructed with builder:
```java
    final FeasibilityCheck feasibilityCheck = aNewFeasibilityCheck()
            .client(aNewClient().firstName("Ivan").surname("Petrov").family("Ivanov").egn("1234567890").telephone("08888888").email("test@company.com").clientType(ClientType.PERSON).build())
            .serviceAddress(aNewAddress().city(new City("city", "PC"))
                    .residentialArea(new ResidentialArea("type", "name"))
                    .street(new Street("street", "type")).streetNumber("20").build())
            .creationInfo(aNewCreationInfo().creatorId(currentUser.user().getEidCode()).createdBy(currentUser.user().getEmail())
                    .creatorFullName(currentUser.user().getFullNameLine()).creationDate(currentUser.time().toDate()).office(currentUser.office()).build())
            .build();
```

The feasibilityCheck has 3 objs in it - client, adress, creationInfo.

**Comment by panayotkulchev** [comment]
After fix both PRs you get LGTM and can mark the XML topic as passed.

File: src/com.clouway.xml/Address.java (hunk 1 above)
Comment (KrasimirRaikov) [inline src/com.clouway.xml/Address.java]: fixed.

File: src/com.clouway.xml/SaxHandler.java (hunk 2 above)
Comment (KrasimirRaikov) [inline src/com.clouway.xml/SaxHandler.java]: done.

File: src/com.clouway.xml/SaxHandler.java (hunk 3 above)
Comment (KrasimirRaikov) [inline src/com.clouway.xml/SaxHandler.java]: done.

File: src/com.clouway.xml/SaxHandler.java (hunk 4 above)
Comment (KrasimirRaikov) [inline src/com.clouway.xml/SaxHandler.java]: rearenged

**Pull request merged**